src/plugins/azure/auth.ts

```diff
@@ -10,47 +10,37 @@ You should have received a copy of the GNU General Public License along with @p0
 **/
 import { print2 } from "../../drivers/stdio";
 import { exec } from "../../util";
-import { KnownError } from "./types";
-
-const knownLoginErrors: KnownError[] = [
-  {
-    pattern:
-      /WARNING: A web browser has been opened at .+ Please continue the login in the web browser.+/,
-    message: "Login attempt was cancelled. Please try again.",
-  },
-];
-
-const knownAccountSetErrors: KnownError[] = [
-  {
-    pattern: /ERROR: The subscription of '.+' doesn't exist in cloud '.+'.+/,
-    message: "Failed to set the active Azure subscription. Please try again.",
-  },
-];
-
-const normalizeAzureCliError = (
-  error: any,
-  normalizedErrors: KnownError[],
-  options: { debug?: boolean }
-) => {
-  if (options.debug) {
-    print2(error);
-  }
-  for (const { pattern, message } of normalizedErrors) {
-    if (pattern.test(error.stderr)) {
-      throw message;
-    }
-  }
-  throw error;
-};
-
-export const azLoginCommand = () => ({
+import { AzureSshRequest } from "./types";
+
+const SUBSCRIPTION_NOT_FOUND_PATTERN =
+  /ERROR: The subscription of '.+' doesn't exist in cloud '.+'.+/;
+const FAILED_TO_RESOLVE_TENANT_PATTERN = /Failed to resolve tenant '.+'/;
+const LOGIN_ATTEMPT_CANCELLED_PATTERN =
+  /WARNING: A web browser has been opened at .+ Please continue the login in the web browser.+/;
+export const AUTHORIZATION_FAILED_PATTERN =
+  /The client '.+' with object id '.+' does not have authorization to perform action '.+' over scope '.+' or the scope is invalid. If access was recently granted, please refresh your credentials/;
+export const USER_NOT_IN_CACHE_PATTERN =
+  /Exception in handling client: User '.+' does not exist in MSAL token cache./;
+export const CONTACT_SUPPORT_MESSAGE =
+  "If the issue persists, please contact support@p0.dev.";
+export const NASCENT_ACCESS_GRANT_MESSAGE =
+  "If access was recently granted, please try again in a few minutes.";
+export const ABORT_AUTHORIZATION_FAILED_MESSAGE = `Your Microsoft Token Cache is out of date. Run 'az account clear' and 'az login' to refresh your credentials. ${CONTACT_SUPPORT_MESSAGE}`;
+
+export const azLoginCommand = (tenantId: string) => ({
   command: "az",
-  args: ["login"],
+  args: [
+    "login",
```

Comment:
This is a small stop-gap to help prevent users from picking the wrong account. When the login prompt appears it'll only show users account's that belong to that tenant.